extensions/devservices/common/src/main/java/io/quarkus/devservices/common/ContainerLocator.java

```diff
@@ -46,7 +46,8 @@ public ContainerLocator(String devServiceLabel, int port) {
                     .flatMap(container -> getMappedPort(container, port)
                             .flatMap(containerPort -> Optional.ofNullable(containerPort.getPublicPort())
                                     .map(port -> {
-                                        final ContainerAddress containerAddress = new ContainerAddress(containerPort.getIp(),
+                                        final ContainerAddress containerAddress = new ContainerAddress(
+                                                DockerClientFactory.instance().dockerHostIpAddress(),
```

Comment:
Does that work when you use a custom network?

Reply:
Currently, dev services containers run in custom bridge networks. Here we are looking for port mappings exposed to the host. So it makes sense to return the host IP.